src/care/LoopFuser.h

```diff
@@ -776,6 +816,7 @@ void LoopFuser::registerAction(const char * fileName, int lineNumber, int start,
          if (verbose) {
             printf("%p: Registering action %i type %i with start %i and end %i\n", this, m_action_count, scan_type, start, end);
          }
+         waitIfNeeded();
```

Comment:
This may be more restrictive than necessary, you can wait until just before you assign the result of instantiate to m_cw or m_aw. That could let you enqueue new work in the pool while the old work is still running form the memory owned by the group.

Reply:
I would like to do this in a separate PR so I can more rigorously test the change in isolation.